ext/mixed/js/japanese.js

```diff
@@ -17,18 +17,102 @@
  */
 
 
+const jpHalfWidthCharacterMapping = new Map([
```

Comment:
Do you think a codepoint-based mapping could work here too?

Reply:
It actually seems quite difficult because halfwidth katakana doesn't have separate codepoints for kana with diacritics and the conversion would get off sync:
https://en.wikipedia.org/wiki/Katakana_(Unicode_block)#Block
https://en.wikipedia.org/wiki/Halfwidth_and_Fullwidth_Forms_(Unicode_block)#Block
The ordering is also different.